From fa634883629aab1aff7a27ba10defc666cc886b9 Mon Sep 17 00:00:00 2001
From: Naveen Mamindlapalli <naveenm@marvell.com>
Date: Fri, 17 Apr 2020 20:56:27 +0530
Subject: [PATCH 517/767] octeontx2-bphy-netdev: added full packet dump of mbt
 buffer

commit 296e2cfb4968f411a7e199fa83573193dce691d0 from
git@git.assembla.com:cavium/WindRiver.linux.git

Added full packet dump of mbt buffer for debugging purpose.

Change-Id: Ifecbcc6367c8395ea0852e926873c713c3c9f345
Signed-off-by: Naveen Mamindlapalli <naveenm@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/27110
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
index 6a8acda65967..9d01fc0ef6a8 100644
--- a/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
+++ b/drivers/net/ethernet/marvell/octeontx2/bphy/otx2_rfoe.c
@@ -410,6 +410,12 @@ static void otx2_rfoe_process_rx_pkt(struct otx2_rfoe_ndev_priv *priv,
 	len = (jd_dma_cfg_word_0->block_size) << 2;
 	netif_dbg(priv, rx_status, priv->netdev, "jd rd_dma len = %d\n", len);
 
+	if (unlikely(netif_msg_pktdata(priv))) {
+		netdev_printk(KERN_DEBUG, priv->netdev, "RX MBUF DATA:");
+		print_hex_dump(KERN_DEBUG, "", DUMP_PREFIX_OFFSET, 16, 4,
+			       buf_ptr, len, true);
+	}
+
 	buf_ptr += (ft_cfg->pkt_offset * 16);
 	len -= (ft_cfg->pkt_offset * 16);
 
@@ -455,13 +461,6 @@ static void otx2_rfoe_process_rx_pkt(struct otx2_rfoe_ndev_priv *priv,
 	skb_put(skb, len);
 	skb->protocol = eth_type_trans(skb, netdev);
 
-	if (unlikely(netif_msg_pktdata(priv2))) {
-		netdev_printk(KERN_DEBUG, priv2->netdev, "Rx: skb %pS len=%d\n",
-			      skb, skb->len);
-		print_hex_dump(KERN_DEBUG, "", DUMP_PREFIX_OFFSET, 16, 4,
-			       skb->data, skb->len, true);
-	}
-
 	if (priv2->rx_hw_tstamp_en)
 		skb_hwtstamps(skb)->hwtstamp = ns_to_ktime(tstamp);
 
-- 
2.31.1

